### ISSUE TITLE
feature request: suppress "no longer available"

### ISSUE THREAD

**Issue by anuramat** [stevearc/oil.nvim#207]
### Did you check existing requests?

- [X] I have searched the existing issues

### Describe the feature

When deleting a file that is open, the corresponding buffer is still loaded and you get a "no longer available" message.
It would be nice to have an option that would automatically delete buffers that correspond to deleted files, and save+reopen buffers for the moved files.

### Provide background

Moving/deleting open files is annoying atm.

### Additional details

Got the idea from `tpope/vim-eunuch`s `:Remove`

**Comment by stevearc** [comment]
Yeah, this would be possible. The current behavior intentionally leaves the buffers in-place as sort of an extra protection from data loss, but we could potentially add another config option to remove those buffers. Will add to my backlog

**Comment by windowsrefund** [comment]
Astrocommunity gets this done [here](https://github.com/AstroNvim/astrocommunity/blob/e20b0412539e0aa42ecc97c8230ca4615db662f9/lua/astrocommunity/file-explorer/oil-nvim/init.lua#L21). My lua-fu is too weak so my struggles to piece together something distro-agnostic are real Maybe someone can improve on this?

```
vim.api.nvim_create_autocmd("User", {
  desc = "Close buffers when files are deleted in Oil",
  pattern = "OilActionsPost",
  callback = function(args)
    if args.data.err then return end
    for _, action in ipairs(args.data.actions) do
      if action.type == "delete" then
        local _, path = require("oil.util").parse_url(action.url)
        local bufnr = vim.fn.bufnr(path)
        vim.cmd.bwipeout({ bufnr, bang = true})
      end
    end
  end,
})
```

**Comment by Tejas-H5** [comment]
+1 on this feature. My use case is that when I move a bunch of typescript files, my LSP will automatically update the imports of all the files referring to the files I moved. I then use :wa to save all the buffers, and this results in the prior buffers being re-created. My current workaround is to delete those files again, then close/reopen neovim.

**Comment by alanfzf** [comment]
+1 this behavior is really annoying
